### DISCUSSION TITLE
UPDATE React to 18.0.0

### DISCUSSION THREAD

**Discussion by ezhilkarthi** [eclipse-theia/theia#11206] [General]
Hi,
while I'm trying to update the react to 18.0.0 , Facing a few issues in types. Is there any document available to upgrade react in theia?

`Type 'React.ReactNode' is not assignable to type 'import("D:/Project/xxx/frontend/node_modules/@types/react/index").ReactNode'.`

**Comment by vince-fugnitto** [comment]
Hi @ezhilkarthi 

> Is there any document available to upgrade react in theia?

There is no such documentation in Theia itself, you might want to look at React itself (migration to newer versions).
Upgrading React should happen across the framework, not necessarily in your downstream application. In addition, if you were to use two different versions of React it would likely be problematic (as highlighted in their migration docs).

How are you performing the upgrade?

**Reply by vince-fugnitto** [reply]
Some additional info, we use `react-virtualized` (which I believe throws the errors currently).
It hasn't released a new version yet to support `react 17` and `react 18` (https://github.com/bvaughn/react-virtualized/pull/1740#issuecomment-1118571314).
